astropy/erfa/core.py.templ

```diff
@@ -66,7 +66,23 @@ class ErfaWarning(AstropyUserWarning):
 
 STATUS_CODES = {}  # populated below before each function that returns an int
 
+# This is a hard-coded list of status codes that need to be remapped,
+# such as to turn errors into warnings.
+STATUS_CODES_REMAP = {
+    'cal2jd': {-3: 3}
+}
+
+
 def check_errwarn(statcodes, func_name):
+    # Remap any errors into warnings in the STATUS_CODES_REMAP dict.
+    if func_name in STATUS_CODES_REMAP:
+        mapping = STATUS_CODES_REMAP[func_name]
+        for i in range(len(statcodes)):
```

Comment:
Sure -- I wasn't quite sure how to achieve that, and didn't give it much thought because the max length of stat codes is 3...